Formula/elasticsearch.rb

```diff
@@ -115,13 +120,41 @@ def plist
     server.close
 
     system "#{bin}/elasticsearch-plugin", "list"
-    pid = "#{testpath}/pid"
+    pid = testpath/"pid"
     begin
       system "#{bin}/elasticsearch", "-d", "-p", pid, "-Epath.data=#{testpath}/data", "-Ehttp.port=#{port}"
       sleep 10
       system "curl", "-XGET", "localhost:#{port}/"
     ensure
-      Process.kill(9, File.read(pid).to_i)
+      Process.kill(9, pid.read.to_i)
+    end
+
+    require "socket"
+
+    server = TCPServer.new(0)
+    port = server.addr[1]
+    server.close
+
+    (testpath/"config/elasticsearch.yml").write <<~EOS
+      path.data: #{testpath}/data
+      path.logs: #{testpath}/logs
+      node.name: test-es-path-conf
+      http.port: #{port}
+    EOS
+
+    cp etc/"elasticsearch/jvm.options", "config"
+    cp etc/"elasticsearch/log4j2.properties", "config"
+
+    ENV["ES_PATH_CONF"] = testpath/"config"
+    pid = testpath/"pid"
+    begin
+      system "#{bin}/elasticsearch", "-d", "-p", pid
+      sleep 10
+      system "curl", "-XGET", "localhost:#{port}/"
+      output = shell_output("curl -s -XGET localhost:#{port}/_cat/nodes")
+      assert_match /test-es-path-conf/, output
```

Comment:
If we're not actually checking against a regex here just using `""` instead of `//` would be nice. ILZ can overrule as desired 🙈.

Reply:
yes I already said that here https://github.com/Homebrew/homebrew-core/pull/20735#discussion_r151705124 but perhaps that here but I think @jasontedor missed that part

Reply:
Ah, apologies. I clicked through a few of the review bubbles but not all of them, evidently 😅.

Reply:
I pushed.

Reply:
Yes, I missed it on the first round, I'm sorry for missing it. I appreciate the second ping. 😇